internal/providers/terraform/aws/instance.go

```diff
@@ -31,16 +35,40 @@ func NewInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 		purchaseOption = "spot"
 	}
 
+	var instanceType, ami, cpuCredits string
+	var ebsOptimized, monitoring bool
+
+	launchTemplateRefs := d.References("launch_template.#.id")
+	if len(launchTemplateRefs) == 0 {
+		launchTemplateRefs = d.References("launch_template.#.name")
+	}
```

Comment:
It can be defined either with `id` or `name` https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#launch-template-specification:
> The launch_template block supports the following:
>
>   `id` - The ID of the launch template. Conflicts with `name`.
>   `name` - The name of the launch template. Conflicts with `id`.